packages/build/src/plugins/mutations.js

```diff
@@ -23,13 +23,13 @@ const applyMutation = function ({ priorityConfig, keys, value, event }) {
     throwValidationError(`"netlifyConfig.${propName}" is read-only.`)
   }
 
-  const { lastEvent, handler } = MUTABLE_PROPS[propName]
+  const { lastEvent, denormalize } = MUTABLE_PROPS[propName]
   validateEvent(lastEvent, event, propName)
 
-  set(priorityConfig, keysString, value)
-
-  if (handler !== undefined) {
-    handler(priorityConfig, value, keys)
+  if (denormalize === undefined) {
+    set(priorityConfig, keysString, value)
+  } else {
+    denormalize(priorityConfig, value)
```

Comment:
We inverse the logic here. Instead of normalizing the configuration (emulating what `@netlify/config` does), we need to denormalize it (inverting what `@netlify/config` does) since `priorityConfig` is passed as input to `@netlify/config` which expects the non-normalized format.